libcst/codemod/commands/convert_type_comments.py

```diff
@@ -0,0 +1,163 @@
+
+
+class ConvertTypeComments(VisitorBasedCodemodCommand):
+    """
+    Codemod that converts type comments, as described in
+    https://www.python.org/dev/peps/pep-0484/#type-comments,
+    into PEP 526 annotated assignments.
+    """
+
+    TYPE_COMMENT_REGEX: Pattern[str] = re.compile("# *type: *[^ ]+")
+    TYPE_IGNORE_PREFIX: str = "# type: ignore"
+
+    _should_strip_type_comments: bool = False
+
+    def __init__(self, context: CodemodContext) -> None:
+        if (sys.version_info.major, sys.version_info.minor) < (3, 8):
+            # The ast module did not get `type_comments` until Python 3.7.
+            # In 3.6, we should error than silently running a nonsense codemod.
+            #
+            # NOTE: it is possible to use the typed_ast library for 3.6, but
+            # this is not a high priority right now. See, e.g., the
+            # mypy.fastparse module.
+            raise NotImplementedError(
+                "You are trying to run ConvertTypeComments on a "
+                + "python version without type comment support. Please "
+                + "try using python 3.8+ to run your codemod."
+            )
+        super().__init__(context)
+
+    def _is_type_comment(self, comment: Optional[cst.Comment]) -> bool:
+        return (
+            comment is not None
+            and self.TYPE_COMMENT_REGEX.match(comment.value) is not None
+            and not comment.value.startswith(self.TYPE_IGNORE_PREFIX)
+        )
+
+    def _should_strip_comment(self, comment: Optional[cst.Comment]) -> bool:
+        return self._should_strip_type_comments and self._is_type_comment(comment)
+
+    def _convert_Assign(
+        self,
+        assign: cst.Assign,
+        type_comment: str,
+    ) -> Union[cst.AnnAssign, cst.Assign]:
+        if len(assign.targets) != 1:
+            return assign
+        return cst.AnnAssign(
+            target=assign.targets[0].target,
+            annotation=_convert_annotation(raw=type_comment),
+            value=assign.value,
+            semicolon=assign.semicolon,
+        )
+
+    def visit_SimpleStatementLine(
+        self,
+        node: cst.SimpleStatementLine,
+    ) -> None:
+        # manage the state used for determining comments to delete
+        self._should_strip_type_comments = True
+
+    def leave_SimpleStatementLine(
+        self,
+        original_node: cst.SimpleStatementLine,
+        updated_node: cst.SimpleStatementLine,
+    ) -> cst.SimpleStatementLine:
+        """
+        Convert any SimpleStatementLine containing an Assign with a
+        type comment into a one that uses a PEP 526 AnnAssign.
+        """
+        # manage the state used for determining comments to delete
+        self._should_strip_type_comments = False
+        # determine whether to apply an annotation
+        body = updated_node.body
+        if len(body) != 1:
+            return updated_node
+        statement = body[0]
```

Comment:
Not sure if this is not going to work if you have multiple statements on the same line. E.g.
```
foo(); x = 42  # type: int
```
The type comment should apply to the last stmt, not the first one.